R/modules.R

```diff
@@ -121,23 +121,40 @@ createSessionProxy <- function(parentSession, ...) {
 #'
 #' @export
 moduleServer <- function(id, module, session = getDefaultReactiveDomain()) {
+  if (inherits(session, c("MockShinySession"))) session$isModuleServer <- TRUE
```

Comment:
Instead of adding `session$isModuleServer` here and checking it in `callModule`, I think it would be clearer if the modifications to `module` were done in this function. You could skip the call to `callModule` in the case where `session` is a `MockShinySession`.

I also still feel a weird about embedding this test-related code into the `moduleServer` or `callModule` functions, but I don't see a better way to do it at the moment.